src/main/scala/lms/transformation/tensor.scala

```diff
@@ -67,7 +70,11 @@ trait FixedSizeTensorFrontEnd extends Base with PrimitiveOps with ArrayOps {
   def TENSOR(shape: Seq[Int], array: ARRAY)(implicit __pos: SourceContext): TENSOR = {
     TENSOR(Adapter.g.reflect("tensor", C(shape), array.x)).withSrcType(__pos, array.et)
   }
+}
 
+trait FixedSizeTensorOps extends Base with PrimitiveOps with ArrayOps {
+  import ArrayTypeLess._
+  import FixedSizeTensorTypeLess._
```

Comment:
these 2 imports are expected because we like the idea of `typed frontend shallowly wraps typeless frontend` to reuse the graph construction logic